emacs-nushell.el

```diff
@@ -12,31 +15,76 @@
 
 (defun nu/format-rows (rows)
   (let ((id -1))
-    (incf id)
     (mapcar (lambda (row)
-              (list id (vector (mapcar #'nu/prepare-column-value row))))
+              (list (cl-incf id)
```

Comment:
`incf` stems from the deprecated `cl.el` package. In Emacs 24.3, the default entry point to the common lisp emulation library  was changed into `cl-lib.el`, which prefixes all `cl` functions with `cl-`. The `cl-incf` variant is the real deal nowadays, `incf` is only an alias.

From `(info "(cl) Organization")`

> The file ‘cl-lib.el’ includes all necessary ‘autoload’ commands for the functions and macros in the other three files.  All you have to do is ‘(require 'cl-lib)’, and ‘cl-lib.el’ will take care of pulling in the other files when they are needed.
>
> …
>
>There is another file, ‘cl.el’, which was the main entry point to this package prior to Emacs 24.3.  **Nowadays, it is replaced by ‘cl-lib.el’.**  The two provide the same features (in most cases), but use different function names (in fact, ‘cl.el’ mainly just defines aliases to the ‘cl-lib.el’ definitions).  Where ‘cl-lib.el’ defines a function called, for example, ‘cl-incf’, ‘cl.el’ uses the same name but without the ‘cl-’ prefix, e.g., ‘incf’ in this example.  There are a few exceptions to this.  [emphasis mine]

From NEWS.24.3 (`C-u C-h C-n 24.3 RET`):

>  ### CL's main entry is now (require 'cl-lib).
> `cl-lib' is like the old `cl' except that it uses the namespace cleanly; i.e., all its definitions have the "cl-" prefix (and internal definitions use the "cl--" prefix).
>
> If `cl' provided a feature under the name `foo', then `cl-lib' provides it under the name `cl-foo' instead; with the exceptions of the few `cl' definitions that had to use `foo*' to avoid conflicts with pre-existing Elisp entities.  These have been renamed to `cl-foo' rather than `cl-foo*'.

> **The old `cl' is now deprecated and is mainly just a bunch of aliases that provide the old, non-prefixed names.** [emphasis mine]

Reply:
Eh, that wasn't an answer to the question, sorry 'bout that.

It is the same as `incf`, so there is no direct benefit. However, if you publish `emacs-nushell` then you probably want to use the `cl-lib.el` variants instead of `cl.el` to keep the global namespace clean :)

Reply:
That was a legitimate answer in my view :)  Thanks!